### ISSUE TITLE
Doprava zasilkovna.cz + zasielkovna.sk

### ISSUE THREAD

**Issue by lukasprelovsky** [pavelevap/ceske-sluzby#15]
* WooCommerce: Doprava (zasilkovna.cz) - CZ i SK

http://www.zasilkovna.cz/pristup-k-pobockam

**Comment by vayCZ** [comment]
Případně by nebylo na škodu přidat možnost automatického podání (API: http://www.zasilkovna.cz/popis-api/)

**Comment by vayCZ** [comment]
Aktuálně se pokouším přidat podporu Zásilkovny.cz, ale nevím, proč se mi nechce zobrazit seznam poboček v pokladně.

**Comment by pavelevap** [comment]
Ano, automatické podání by bylo fajn :-)

@vayCZ: Máte někde nějaký konkrétní kód?

**Comment by viktorjezekcz** [comment]
@pavelevap: Myslíte ke způsobu dopravy? Zatím jsem zkoušel vycházet již z vašeho hotového řešení pro Uloženku, ale nedaří se. Použil jsem JS API nabízené na stránkách Zásilkovny. Nový způsob dopravy se sice v košíku objeví, ale když na to kliknu, tak se neobjeví rozevírací seznam se seznamem poboček.

**Comment by pavelevap** [comment]
Zkuste někam uložit konkrétní kód (třeba nahoře je odkaz na Gist), zkusím se na to podívat...

**Comment by viktorjezekcz** [comment]
https://gist.github.com/viktorjezekcz/2dbcc1098068596f07d8 zde je celý kód ceske-sliuzby.php. Kód pro zobrazování zásilkovny v pokladně je úplně na konci.

**Comment by pavelevap** [comment]
@viktorjezekcz: Ještě by tam měl být načítaný soubor `includes/class-ceske-sluzby-zasilkovna.php`?

**Comment by vayCZ** [comment]
@pavelevap: Ano, má se tam načítat, ale pokud dobře chápu, atk ten má nastarosti jen administraci, ne?

**Comment by viktorjezekcz** [comment]
Právě jsem objevil, svůj druhý účet na github :D

**Comment by viktorjezekcz** [comment]
@pavelevap: Ano, má se tam načítat, ale pokud dobře chápu, atk ten má nastarosti jen administraci, ne?

Napíšu to znovu, aby to bylo evidováno pod mým aktuálním účtem. Ten předchozí je už smazaný, proto je tam jako uživatel ghost.

**Comment by pavelevap** [comment]
@viktorjezekcz: To sice ano, ale je tam načítán konkrétní skript `class-ceske-sluzby-zasilkovna.php`, který s tím může souviset (respektive se nemusí pobočky zobrazovat právě kvůli chybějícímu nastavení).

**Comment by viktorjezekcz** [comment]
@pavelevap: Tak tedy, skript `class-ceske-sluzby-zasilkovna.php`: https://gist.github.com/viktorjezekcz/55467054292a63201553

**Comment by viktorjezekcz** [comment]
Jak to tedy vypadá s implementací zásilkovny?

**Comment by pavelevap** [comment]
Zásilkovna se nám docela rozmáhá, asi nastal čas uvolnit skript mezi uživatele :-) Zájemci mohou otestovat, výběr a ukládání poboček by měl být funkční, automatické odesílání přímo do Zásilkovny se objeví brzy také (musím z toho udělat obecný plugin s nastavením).

Co ještě chybí?
- [ ] Je takto načítaný JavaScript na správném místě, možná by to chtělo přesunout někam jinam než `wp_footer`?
- [ ] Zkontrolovat funkčnost pluginu WOOCS – Currency Switcher for WooCommerce.
- [ ] Plugin WooCommerce Pay for Payment už na nic nepotřebujeme, takže zmigrovat nastavení, pokud ho někdo používal u jiných druhů dopravy (Uloženka, DPD).
- [ ] Zkontrolovat funkčnost localStorage ([link](https://stackoverflow.com/questions/16427636/check-if-localstorage-is-available)).
- [ ] Načítat různé pobočky pro různé země (CZ a SK), a to podle informací o zákazníkovi?
- [ ] Doplnit některá nastavení z JS API Zásilkovny.
- [ ] Odesílat informace o objednávkách přímo do Zásilkovny.
